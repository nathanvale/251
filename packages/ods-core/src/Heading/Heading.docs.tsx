import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { Box } from "../Box/Box";
import { Stack } from "../Stack/Stack";
import { HeadingProps, Heading } from "./Heading";

export const headingPropDescriptions: Partial<Record<
  keyof HeadingProps,
  string
>> = {
  component: "Used to change the semantic meaning for SEO",
  truncate:
    "This will add an ellipsis to the end of the heading if the heading drops onto more than one line",
};

export const docs: ComponentDocs<HeadingProps> = {
  category: "Content",
  componentName: "Heading",
  description:
    "Heading is used to define HTML headings. The variant h1 is the most important heading. The variant h4 defines the least important heading.",
  propDescriptions: { ...headingPropDescriptions },
  specialOptionalProps: [
    {
      name: "weight",
      defaultValue: "defaultValue",
      type: {
        label: `"regular" | "medium"`,
        description: `This property only applies to heading variants of "h1" or "h2". Other use cases will throw a typescript error or a runtime error in playroom`,
      },
    },
  ],
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <Heading variant="h1">
          The quick brown fox jumps over the lazy dog
        </Heading>
      ),
    },
    additional: [
      {
        label: "Heading H2",
        Code: () => (
          <Heading variant="h2">
            The quick brown fox jumps over the lazy dog
          </Heading>
        ),
      },
      {
        label: "Heading H3",
        Code: () => (
          <Heading variant="h3">
            The quick brown fox jumps over the lazy dog
          </Heading>
        ),
      },
      {
        label: "Heading H4",
        Code: () => (
          <Heading variant="h4">
            The quick brown fox jumps over the lazy dog
          </Heading>
        ),
      },
      {
        label: "Truncating long headings",
        Code: () => (
          <Box style={{ width: 120 }}>
            <Heading variant="h4" truncate>
              The quick brown fox jumps over the lazy dog
            </Heading>
          </Box>
        ),
      },
      {
        label: "Heading Alignment",
        Code: () => (
          <Stack space="medium">
            <Heading variant="h4">
              The quick brown fox jumps over the lazy dog
            </Heading>
            <Heading variant="h4" align="center">
              The quick brown fox jumps over the lazy dog
            </Heading>
            <Heading variant="h4" align="right">
              The quick brown fox jumps over the lazy dog
            </Heading>
          </Stack>
        ),
      },
      {
        label: "Heading Alignment (responsive)",
        Code: () => (
          <Heading variant="h4" align={["left", "center"]}>
            The quick brown fox jumps over the lazy dog
          </Heading>
        ),
      },
      {
        label: `Applying medium weight`,
        description: `This only applies to heading variants of "h1" or "h2". Other use cases will throw a typescript error or a runtime error in playroom.`,
        Code: () => (
          <Heading variant="h2" weight="medium">
            The quick brown fox jumps over the lazy dog
          </Heading>
        ),
      },
      {
        label: "Changing the semantic meaning for SEO",
        Code: () => (
          <Heading variant="h4" component="h1">
            The quick brown fox jumps over the lazy dog
          </Heading>
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Heading 1",
      Code: () => <Heading variant="h1">Heading 1</Heading>,
    },
    {
      label: "Heading 2",
      Code: () => <Heading variant="h2">Heading 2</Heading>,
    },
    {
      label: "Heading 3",
      Code: () => <Heading variant="h3">Heading 3</Heading>,
    },
    {
      label: "Heading 4",
      Code: () => <Heading variant="h4">Heading 4</Heading>,
    },
  ],
};
