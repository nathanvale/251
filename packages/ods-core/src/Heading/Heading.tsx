import React, { ReactNode } from "react";
import clsx from "clsx";
import {
  HeadingComponentVariants,
  TypographyWeightVariants,
  Heading12Variants,
  Heading34Variants,
  HeadingVariants,
} from "@origin-digital/ods-types";
import {
  useBasekickStyles,
  useToneStyles,
  useStrongStyles,
  useWeightStyles,
} from "@origin-digital/ods-typography";
import {
  useTruncatedContent,
  useCheckTypographyBackground,
} from "../_private/hooks";
import { BoxProps, Box } from "../Box/Box";
import { Text } from "../Text/Text";
import { HeadingContext } from "./HeadingContext";

export interface HeadingBaseProps {
  children: ReactNode;
  align?: BoxProps["textAlign"];
  truncate?: boolean;
  "data-id"?: string;
  component?: HeadingComponentVariants;
  /**
   * Heading font weights can only be applied to h1 and h2 variants.
   * Otherwise a typescript error error will be displayed.
   */
  weight?: Exclude<TypographyWeightVariants, "bold">;
}

type Heading12Props = Omit<HeadingBaseProps, "variant"> & {
  variant: Heading12Variants;
};

type Heading34Props = Omit<HeadingBaseProps, "variant" | "weight"> & {
  variant: Heading34Variants;
};

export type HeadingProps = Heading12Props | Heading34Props;

interface UseHeadingStylesProps {
  variant: Heading12Variants | Heading34Variants;
  weight: Exclude<TypographyWeightVariants, "bold">;
}

function useHeadingStyles(props: UseHeadingStylesProps) {
  const { variant, weight } = props;
  useCheckTypographyBackground();
  const basekickStyles = useBasekickStyles({ variant });
  const toneStyles = useToneStyles({ tone: "neutralDark" });
  const strongStyles = useStrongStyles("bold");
  const weightMap: Record<
    HeadingVariants,
    Exclude<TypographyWeightVariants, "bold">
  > = {
    h1: weight,
    h2: weight,
    h3: "medium",
    h4: "medium",
  };
  const weightStyles = useWeightStyles(weightMap[variant!]);
  return clsx(basekickStyles, toneStyles, strongStyles, weightStyles);
}

export const Heading = (props: HeadingProps) => {
  const {
    children,
    component,
    truncate = false,
    align,
    "data-id": dataId,
    variant,
  } = props;

  let weight: HeadingBaseProps["weight"];
  if (props.variant === "h2" || props.variant === "h1") {
    weight = props.weight || "regular";
  } else {
    // @ts-ignore (ignoring in this case so we can throw a runtime error)
    if (props.weight)
      throw new Error(
        "Heading font weights can only be applied to h1 and h2 variants."
      );
    weight = "medium";
  }
  const headingStyles = useHeadingStyles({ variant, weight });
  const content = useTruncatedContent({ children, truncate, display: "block" });

  if (
    variant !== "h1" &&
    variant !== "h2" &&
    variant !== "h3" &&
    variant !== "h4"
  )
    return (
      <Text tone="critical">
        Please choose a heading variant! ( "h1" | "h2" | "h3" | "h4")
      </Text>
    );

  return (
    <HeadingContext.Provider value={true}>
      <Box
        data-id={dataId}
        className={headingStyles}
        component={component || variant}
        textAlign={align}
      >
        {content}
      </Box>
    </HeadingContext.Provider>
  );
};

Heading.defaultProps = {
  "data-id": "heading",
  truncate: false,
  align: "left",
};

Heading.displayName = "Heading";
