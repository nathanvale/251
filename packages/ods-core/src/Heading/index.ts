export { Heading } from "./Heading";
export { HeadingContext } from "./HeadingContext";
export type HeadingProps = import("./Heading").HeadingProps;
