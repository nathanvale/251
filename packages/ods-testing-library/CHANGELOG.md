# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
