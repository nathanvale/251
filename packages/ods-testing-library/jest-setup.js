import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
