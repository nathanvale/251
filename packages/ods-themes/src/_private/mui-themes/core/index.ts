import createMuiTheme, {
  ThemeOptions,
} from "@material-ui/core/styles/createMuiTheme";

import { flattenPalette } from "../../helpers/flattenPalette";
import { palette } from "./palette";
import { breakpoints } from "./breakpoints";
import { shadows } from "./shadows";
import { getMUITypography } from "./typography";

export const themeOptions: ThemeOptions = {
  colors: flattenPalette(palette),
  palette,
  // Breakpoint
  breakpoints,
  // Shadows
  shadows,
  shape: {
    borderRadius: 0,
  },
  typography: getMUITypography(false),

  // Default props
  props: {
    MuiPaper: {
      square: true,
    },
    MuiTextField: {
      autoComplete: "false",
    },
  },

  // overrides: {
  //
  // }
};

export const coreMuiTheme = createMuiTheme(themeOptions);
