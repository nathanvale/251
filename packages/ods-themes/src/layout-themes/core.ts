import { Space, Theme, Transition } from "@origin-digital/ods-types";

const gridGutterWidth = 16;

const space: Space = [0, 4, 8, 12, 16, 24, 32, 48, 64];
space.none = space[0];
space.xxsmall = space[1];
space.xsmall = space[2];
space.small = space[3];
space.medium = space[4];
space.large = space[5];
space.xlarge = space[6];
space.xxlarge = space[7];
space.xxxlarge = space[8];

const transitions: Transition = {
  fast: "transform .15s ease, opacity .15s ease",
};

export const coreLayoutTheme: Theme = {
  space,
  gridGutterWidth,
  transitions,
  section: {
    maxWidth: {
      xs: 540,
      sm: 540,
      md: 720,
      lg: 960,
      xl: 1140,
    },
  },
};
