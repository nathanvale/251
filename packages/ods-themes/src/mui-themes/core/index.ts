import createMuiTheme, {
  ThemeOptions,
} from "@material-ui/core/styles/createMuiTheme";

import { palette } from "./palette";
import { breakpoints } from "./breakpoints";
import { shadows } from "./shadows";
import { getMUITypography } from "./typography";

export const themeOptions: ThemeOptions = {
  palette,
  // Breakpoint
  breakpoints,
  // Shadows
  shadows,
  shape: {
    borderRadius: 0,
  },
  typography: getMUITypography(false),

  // Default props
  props: {
    MuiPaper: {
      square: true,
    },
    MuiTextField: {
      autoComplete: "false",
    },
  },

  // overrides: {
  //
  // }
};

export const coreMuiTheme = createMuiTheme(themeOptions);
