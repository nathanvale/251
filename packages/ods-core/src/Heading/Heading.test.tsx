import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Box } from "@material-ui/core";
import { Heading } from "./Heading";

test("It can render h1", () => {
  const { container } = render(<Heading variant="h1">Heading H1</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render h2", () => {
  const { container } = render(<Heading variant="h1">Heading H1</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render h3", () => {
  const { container } = render(<Heading variant="h1">Heading H1</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render h4", () => {
  const { container } = render(<Heading variant="h1">Heading H1</Heading>);
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render h1 medium weight", () => {
  const { container } = render(
    <Heading variant="h1" weight="medium">
      Heading H1
    </Heading>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render h2 medium weight", () => {
  const { container } = render(
    <Heading variant="h2" weight="medium">
      Heading H1
    </Heading>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test("It can render heaading truncated", () => {
  const { container } = render(
    <Box style={{ width: 90 }}>
      <Heading variant="h2" truncate>
        Truncated text
      </Heading>
    </Box>
  );
  expect(container.firstChild).toMatchSnapshot();
});
