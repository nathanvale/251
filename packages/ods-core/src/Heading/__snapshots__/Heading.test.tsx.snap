// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It can render h1 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<h1
  class="typography-basekick-1 typography-color-2 typography-strong-3 typography-weight-4 -h1 c0"
  data-id="heading"
>
  Heading H1
</h1>
`;

exports[`It can render h1 medium weight 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<h1
  class="typography-basekick-21 typography-color-22 typography-strong-23 typography-weight-24 -h1 c0"
  data-id="heading"
>
  Heading H1
</h1>
`;

exports[`It can render h2 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<h1
  class="typography-basekick-6 typography-color-7 typography-strong-8 typography-weight-9 -h1 c0"
  data-id="heading"
>
  Heading H1
</h1>
`;

exports[`It can render h2 medium weight 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<h2
  class="typography-basekick-26 typography-color-27 typography-strong-28 typography-weight-29 -h2 c0"
  data-id="heading"
>
  Heading H1
</h2>
`;

exports[`It can render h3 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<h1
  class="typography-basekick-11 typography-color-12 typography-strong-13 typography-weight-14 -h1 c0"
  data-id="heading"
>
  Heading H1
</h1>
`;

exports[`It can render h4 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<h1
  class="typography-basekick-16 typography-color-17 typography-strong-18 typography-weight-19 -h1 c0"
  data-id="heading"
>
  Heading H1
</h1>
`;

exports[`It can render heaading truncated 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
  text-align: left;
}

<div
  class="MuiBox-root MuiBox-root-32"
  style="width: 90px;"
>
  <h2
    class="typography-basekick-33 typography-color-34 typography-strong-35 typography-weight-36 -h2 c0"
    data-id="heading"
  >
    <div
      class="MuiBox-root MuiBox-root-38 typography-truncate-37"
    >
      Truncated text
    </div>
  </h2>
</div>
`;
